pcmanfm/mainwindow.cpp

```diff
@@ -287,12 +287,15 @@ MainWindow::~MainWindow() {
 }
 
 void MainWindow::chdir(Fm::FilePath path) {
-    TabPage* page = currentPage();
-    if(page) {
-        ui.filterBar->clear();
-        page->chdir(path, true);
-        updateUIForCurrentPage();
-    }
+    // wait until queued events are processed
+    QTimer::singleShot(0, [this, path] {
```

Comment:
IMO the `void QTimer::singleShot(int msec, const QObject *context, Functor functor)` overload should be used for the sake of correctness (doc: _If **context** is destroyed before the interval occurs, the method will not be called._)

Reply:
Good point.